### DISCUSSION TITLE
I only get a 404

### DISCUSSION THREAD

**Discussion by IeuanK** [photoprism/photoprism#3798] [General]
Copy of https://github.com/photoprism/photoprism/issues/3797, as apparently PhotoPrism not working is not a bug?  

#### 1. What is not working as documented?

PhotoPrism is not loading. The logs show no errors, but I get a black screen with this image:  

![image](https://github.com/photoprism/photoprism/assets/1075598/e2740930-d001-4561-9314-6eea4d558b17)

Important to note is that it has been working fine for well over a year, and no other services on this Docker node are showing issues.

#### 2. How can we reproduce it?

I have no idea what is going wrong, so I can't provide steps to reproduce  

#### 3. What behavior do you expect?

App loading as usual

#### 4. What could be the cause of your problem?

I can't imagine any reason this would happen, at least not without errors. It is immediately sending back a 404 and the logs show that it's a 404 on '/', so this almost has to be internal.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

With debug mode turned on, the log only shows this:  

time="2023-10-07T14:44:29Z" level=debug msg="server: GET / (404) [3.715894ms]"
time="2023-10-07T14:44:29Z" level=debug msg="server: GET / (404) [2.260027ms]"
time="2023-10-07T14:44:32Z" level=debug msg="server: GET / (404) [2.463227ms]"
time="2023-10-07T14:44:36Z" level=debug msg="server: GET / (404) [2.292991ms]"
time="2023-10-07T14:44:37Z" level=debug msg="server: GET / (404) [2.122208ms]"
time="2023-10-07T14:44:40Z" level=debug msg="server: GET / (404) [2.167047ms]"
time="2023-10-07T14:45:06Z" level=debug msg="server: GET / (404) [2.036ms]"
time="2023-10-07T14:45:09Z" level=debug msg="server: GET / (404) [2.0515ms]"

#### 6. Which software versions do you use?

I use the latest docker tag, the docker host runs Debian.  

MariaDB version as per Adminer is `MySQL version: 5.5.5-10.11.3-MariaDB-1:10.11.3+maria~ubu2204`

I've tried various browsers, with and without extensions. I've tried accessing through NginxPM as well as directly accessing the port.

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...  
No clue what CPU, but it's just a `Genuine Intel(R) CPU  @ 2.10GHz`. Other software meant for x64 processors runs fine.  
  
(b) Physical Memory & Swap Space in GB  
Plenty, over 180GB memory free  

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...  
It's a ZFS pool with 10 drives, again no issue with other software, including software that sees heavy disk usage

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

I normally use a very simple proxy host within Nginx Proxy Manager, but accessing the container directly by exposing the port shows the same result

**Comment by graciousgrey** [comment]
Did something regarding your setup change recently?

**Reply by IeuanK** [reply]
Other than pulling the most recent PhotoPrism image, no..

**Reply by lastzero** [reply]
Maybe your site URL includes a path?

**Reply by IeuanK** [reply]
Nope, it's just a docker container, accessing the IP + exposed port directly makes no difference either

**Reply by lastzero** [reply]
Did you already enable debug or trance to mode to check the logs?

**Reply by IeuanK** [reply]
The logs I pasted above are with debug mode enabled (`PHOTOPRISM_DEBUG=true`). Without that I got no log output when trying to open up PhotoPrism

**Reply by lastzero** [reply]
Try trace logs then and ideally add the output of the `photoprism config` command 👌

**Reply by IeuanK** [reply]
Here's the output of `photoprism config`; https://gist.github.com/IeuanK/05ec86d52dd32510fca13e3f6f963157

How do I get trace logs? I only see the debug option explained in the docs/docker hub

**Reply by lastzero** [reply]
Your site URL is a path, please use a complete URL like https//example.com/ as shown in our docs 👍

**Reply by IeuanK** [reply]
Could you clarify where you are seeing this? I am only using a full hostname, no path.   
Also wondering why this suddenly now stopped working if it isn't configured properly, because I have not changed the config

**Reply by IeuanK** [reply]
I just added http:// in front of the hostname in the env variable `PHOTOPRISM_SITE_URL`, which seems to have fixed it. Still leaves the question why this regression was introduced, because it seems wrong to have to explicitly provide the protocol for a hostname

**Reply by lastzero** [reply]
This is not a regression, but an implementation of a feature request to support hosting on a shared domain, and the format of URLs is unanimously determined as follows:

![Screenshot_20231007_184617](https://github.com/photoprism/photoprism/assets/301686/808be133-3272-46a6-bcc6-c3dd91218a1e)
